### ISSUE TITLE
Output context about the offending annotation on a change rule error

### ISSUE THREAD

**Issue by ewrenn8** [k14s/kapp#178]
**Describe the problem/challenge you have**
I was deploying a few yaml resources and got the error:
```
kapp: Error: Change graph:
  Calculating required deps:
    Expected change rule annotation value to have 4 pieces but was 1
```
This wasn't very helpful for tracking down where in the yaml the issue was.

**Describe the solution you'd like**
I would like to see some context about the offending annotation. Outputting the annotation that caused the error as part of the message is one option.

**Comment by DennisDenuto** [comment]
@ewrenn8 Thanks for creating this issue. We agree that this error message could use some additional context. The annotation name and the 'pieces' that are missing is a nice start.

Can you please provide the 'yaml resources' that caused this error so that we can reproduce it on our end?

We are going to create a story in our backlog and will prioritize accordingly.

**Comment by ewrenn8** [comment]
Hey @DennisDenuto,

I don't have the exact yaml I was using, but this should be enough to repro. It was a copy paste error where I forgot to update the annotation from a rule to a group, but didn't know where I had made the error.
```yaml
---
apiVersion: v1
kind: Service
metadata:
  name: server-svc
  annotations:
    kapp.k14s.io/change-rule: "upsert after upserting server"
spec:
  selector:
    app: server
  ports:
  - port: 443
    targetPort: 443
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: server
  annotations:
    kapp.k14s.io/change-rule: "server"
spec:
  replicas: 1
  selector:
    matchLabels:
      app:  server
  template:
    metadata:
      labels:
        app: server
    spec:
      containers:
      - name: server
        image: nginx
        ports:
        - containerPort: 443
```

**Comment by cari-lynn** [comment]
Thanks @ewrenn8

**Comment by renuy** [comment]
Better error message  - "Template of change-rule is : "

**Comment by renuy** [comment]
available in release v0.40.0
